include/seqan3/alphabet/aminoacid/concept.hpp

```diff
@@ -6,47 +6,124 @@
 // -----------------------------------------------------------------------------------------------------
 
 /*!\file
+ * \author Hannes Hauswedell <hannes.hauswedell AT fu-berlin.de>
  * \author Joshua Kim <joshua.kim AT fu-berlin.de>
  * \brief Provides seqan3::aminoacid_alphabet.
  */
 
 #pragma once
 
+#include <type_traits>
+
 #include <seqan3/alphabet/concept.hpp>
 
 // ============================================================================
-// concept
+// aminoacid_empty_base
 // ============================================================================
 
 namespace seqan3
 {
 
-/*!\brief Identifies amino acid alphabets.
- * \implements seqan3::unary_type_trait
+/*!\brief This is an empty base class that can be inherited from to to make types model seqan3::aminoacid_concept.
+ * \ingroup aminoacid
+ * \see seqan3::enable_aminoacid
+ */
+struct aminoacid_empty_base
+{};
+
+} // namespace seqan3
+
+// ============================================================================
+// enable_aminoacid
+// ============================================================================
+
+namespace seqan3::detail::adl_only
+{
+
+//!\brief Poison-pill overload to prevent non-ADL forms of unqualified lookup.
+template <typename ...args_t>
+void enable_aminoacid(args_t ...) = delete;
+
+//!\brief Customisation point dispatcher for seqan3::enable_aminoacid.
+struct enable_aminoacid_fn
+{
+public:
+    // explicit customisation
+    SEQAN3_CPO_IMPL(2, std::bool_constant<seqan3::custom::alphabet<strip_type_identity_t<t>>::enable_aminoacid>::value)
+    // ADL
+    SEQAN3_CPO_IMPL(1, std::bool_constant<enable_aminoacid(t{})>::value)
+    // default: derived from base class or not (valid for any type)
+    SEQAN3_CPO_IMPL(0, (std::is_base_of_v<seqan3::aminoacid_empty_base, strip_type_identity_t<t>>))
+
+    //!\brief Main dispatching function.
+    template <typename alph_t>
+    static constexpr bool impl_main() noexcept
```

Comment:
Well since the interface that we expose is a trait interface and not a function object, we don't need to create a functor. We are just using the static members to the stuff and store the result in the trait variable.

Reply:
why not call this function `dispatch`? I would prefer this about impl_main as main is a quite reserved name with some special meanings. I know this what you like to capture but dispatch sounds more like the actual thing. At least to me

Reply:
but it is not resolved?